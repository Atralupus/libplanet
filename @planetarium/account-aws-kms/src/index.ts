export { AwsKmsKeyStore } from "./AwsKmsKeyStore.js";
export { type AwsKmsKeyId } from "./AwsKmsKeyId.js";
export { AwsKmsAccount } from "./AwsKmsAccount.js";
export { type AwsKmsMetadata } from "./AwsKmsMetadata.js";
export { KMSClient } from "@aws-sdk/client-kms";
