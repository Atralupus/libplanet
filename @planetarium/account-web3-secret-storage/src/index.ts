export { KeyId } from "./KeyId.js";
export { PassphraseEntry } from "./PassphraseEntry.js";
export { TtyPassphraseEntry } from "./TtyPassphraseEntry.js";
export {
  type DecryptionOptions,
  WeakPrivateKeyError,
  Web3Account,
  type Web3KeyObject,
} from "./Web3Account.js";
export { getDefaultWeb3KeyStorePath, Web3KeyStore } from "./Web3KeyStore.js";
