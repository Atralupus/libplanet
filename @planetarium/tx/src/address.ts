import { Key, Value } from "@planetarium/bencodex";
import { compareUint8Array } from "./binary.js";

export type Address = Uint8Array; // TODO: proper type definition

export function encodeAddress(address: Address): Key {
  if (address.length !== 20) {
    throw new TypeError(
      `Address must be 20 bytes long, but got ${address.length} bytes.`,
    );
  }
  return address;
}

export function encodeAddressSet(addresses: Set<Address>): Value {
  const array: Address[] = [];
  addresses.forEach((addr) => array.push(addr));
  array.sort(compareUint8Array);
  return array.map(encodeAddress);
}
