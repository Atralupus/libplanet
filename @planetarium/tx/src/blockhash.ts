import { Value } from "@planetarium/bencodex";

export type BlockHash = Uint8Array; // TODO: proper type definition

export function encodeBlockHash(blockHash: BlockHash): Value {
  if (blockHash.length !== 32) {
    throw new TypeError(
      `BlockHash must be 32 bytes long, but got ${blockHash.length} bytes.`,
    );
  }
  return blockHash;
}
