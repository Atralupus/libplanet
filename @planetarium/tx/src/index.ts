export { type Address, encodeAddress } from "./address";
export {
  type Currency,
  encodeCurrency,
  type FungibleAssetValue,
  getCurrencyHash,
  getMajorUnit,
  getMinorUnit,
  getSign,
} from "./assets";
export {
  encodeTransfer,
  type Transfer,
} from "./action";
