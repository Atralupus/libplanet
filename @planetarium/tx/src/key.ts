import { Value } from "@planetarium/bencodex";

export type PublicKey = Uint8Array; // TODO: proper type definition

export function encodePublicKey(publicKey: PublicKey): Value {
  if (publicKey.length < 1) {
    throw new TypeError("Public key must not be empty.");
  }
  return publicKey;
}
