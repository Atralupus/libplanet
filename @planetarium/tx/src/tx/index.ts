export type { TxMetadata } from "./metadata.js";
export { encodeTxMetadata } from "./metadata.js";
export type {
  UnsignedTxWithSystemAction,
  UnsignedTxWithCustomActions,
} from "./unsigned.js";
export type { SignedTx } from "./signed.js";
export {
  encodeUnsignedTxWithSystemAction,
  encodeUnsignedTxWithCustomActions,
} from "./unsigned.js";
export { signTx, encodeSignedTx } from "./signed.js";
