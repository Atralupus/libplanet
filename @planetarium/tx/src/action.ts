import { Encodable } from "bencodex";
import { Address, encodeAddress } from "./address";
import { encodeCurrency, FungibleAssetValue } from "./assets";

export interface Transfer {
  recipient: Address;
  amount: FungibleAssetValue;
}

export function encodeTransfer(action: Transfer): Encodable {
  return {
    recipient: encodeAddress(action.recipient),
    currency: encodeCurrency(action.amount.currency),
    amount: action.amount.rawValue,
  };
}
