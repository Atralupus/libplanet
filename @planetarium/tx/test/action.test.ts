import { expect, test } from "vitest";
import { Transfer } from "../src/action";
import { Address } from "../src/address";
import { Currency } from "../src/assets";
import { fromHex } from "./hex";

const FOO: Currency = {
  ticker: "FOO",
  decimalPlaces: 2,
  minters: null,
  totalSupplyTrackable: true,
  maximumSupply: null,
};
const addressA: Address = fromHex("D6D639DA5a58A78A564C2cD3DB55FA7CeBE244A9");

test("encodeTransfer", () => {
  const transfer: Transfer = {
    recipient: addressA,
    amount: {
      rawValue: 12500n,
      currency: FOO,
    }
  };
  expect(transfer).toMatchSnapshot();
});
