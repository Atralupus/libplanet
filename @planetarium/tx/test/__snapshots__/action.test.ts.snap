// Vitest Snapshot v1

exports[`encodeTransfer 1`] = `
{
  "amount": {
    "currency": {
      "decimalPlaces": 2,
      "maximumSupply": null,
      "minters": null,
      "ticker": "FOO",
      "totalSupplyTrackable": true,
    },
    "rawValue": 12500n,
  },
  "recipient": Uint8Array [
    214,
    214,
    57,
    218,
    90,
    88,
    167,
    138,
    86,
    76,
    44,
    211,
    219,
    85,
    250,
    124,
    235,
    226,
    68,
    169,
  ],
}
`;
